tracing-subscriber/src/fmt/format/mod.rs

```diff
@@ -1526,23 +1536,60 @@ pub(super) mod test {
             .with_timer(MockTime);
         let expected = "fake time tracing_subscriber::fmt::format::test: hello\n";
 
-        run_test(subscriber, make_writer, expected);
+        assert_info_hello(subscriber, make_writer, expected);
     }
 
-    #[cfg(feature = "ansi")]
-    fn test_ansi(is_ansi: bool, expected: &str) {
+    #[test]
+    fn with_thread_ids() {
         let make_writer = MockMakeWriter::default();
         let subscriber = crate::fmt::Collector::builder()
             .with_writer(make_writer.clone())
-            .with_ansi(is_ansi)
+            .with_thread_ids(true)
+            .with_ansi(false)
+            .with_timer(MockTime);
+        let expected = "fake time  INFO ThreadId() tracing_subscriber::fmt::format::test: hello\n";
+
+        assert_info_hello_ignore_numeric(subscriber, make_writer, expected);
+    }
+
+    #[test]
+    fn pretty_default() {
+        let make_writer = MockMakeWriter::default();
+        let subscriber = crate::fmt::Collector::builder()
+            .pretty()
+            .with_writer(make_writer.clone())
+            .with_ansi(false)
             .with_timer(MockTime);
-        run_test(subscriber, make_writer, expected)
+        let expected = r#"  fake time  INFO tracing_subscriber::fmt::format::test: hello
+    at tracing-subscriber/src/fmt/format/mod.rs:
+
+"#
+        .replace('/', &std::path::MAIN_SEPARATOR.to_string());
```

Comment:
what if, instead of using `replace` to fix the path separator, we just used the `file!()` macro, like this:
```suggestion
        let expected = concat!(
            "  fake time  INFO tracing_subscriber::fmt::format::test: hello\n",
            "    at ",
            file!(),
            ":\n"
        );
```

this way, we will always get the actual path to the file as understood by Cargo.

Reply:
oh thats nice! (hopefully it does what we expect on windows)